docs/design/AnoncredsW3CCompatibility.md

```diff
@@ -0,0 +1,286 @@
+
+- W3C Credential Create
+
+**NOTE: There has been some changes to _encoding of attribute values_ for creating a credential, so we have to be adjust to the new changes.**
+
+```
+async def create_credential(
+        self,
+        credential_offer: dict,
+        credential_request: dict,
+        credential_values: dict,
+    ) -> str:
+...
+...
+  try:
+    credential = await asyncio.get_event_loop().run_in_executor(
+        None,
+        lambda: W3CCredential.create(
+            cred_def.raw_value,
+            cred_def_private.raw_value,
+            credential_offer,
+            credential_request,
+            raw_values,
+            None,
+            None,
+            None,
+            None,
+        ),
+    )
+...
+```
+
+- W3C Credential Request
+
+```
+async def create_w3c_credential_request(
+        self, credential_offer: dict, credential_definition: CredDef, holder_did: str
+    ) -> Tuple[str, str]:
+...
+...
+try:
+  secret = await self.get_master_secret()
+  (
+      cred_req,
+      cred_req_metadata,
+  ) = await asyncio.get_event_loop().run_in_executor(
+      None,
+      W3CCredentialRequest.create,
+      None,
+      holder_did,
+      credential_definition.to_native(),
+      secret,
+      AnonCredsHolder.MASTER_SECRET_ID,
+      credential_offer,
+  )
+...
+```
+
+- W3C Credential Present
+
+```
+async def create_w3c_presentation(
+        self,
+        presentation_request: dict,
+        requested_credentials: dict,
+        schemas: Dict[str, AnonCredsSchema],
+        credential_definitions: Dict[str, CredDef],
+        rev_states: dict = None,
+    ) -> str:
+...
+...
+  try:
+    secret = await self.get_master_secret()
+    presentation = await asyncio.get_event_loop().run_in_executor(
+        None,
+        Presentation.create,
+        presentation_request,
+        present_creds,
+        self_attest,
+        secret,
+        {
+            schema_id: schema.to_native()
+            for schema_id, schema in schemas.items()
+        },
+        {
+            cred_def_id: cred_def.to_native()
+            for cred_def_id, cred_def in credential_definitions.items()
+        },
+    )
+...
+```
+
+#### Converting an already issued legacy anoncreds to W3C format(vice versa)
+
+In this case, we can use `to_w3c` method of `Credential` class to convert from legacy to w3c and `to_legacy` method of `W3CCredential` class to convert from w3c to legacy.<br>
+
+We could call `to_w3c` method like this:
+
+```
+w3c_cred = Credential.to_w3c(cred_def)
+```
+
+and for `to_legacy`:
+
+```
+legacy_cred = W3CCredential.to_legacy()
+```
+
+We don't need to input any parameters to it as it in turn calls `Credential.from_w3c()` method under the hood
+
+### How a W3C credential is stored in the wallet.
+
+Storing a credential in the wallet is somewhat dependent on the kinds of metadata that are relevent. The metadata mapping between the W3C credential and an anoncreds credential is not fully clear yet.
```

Comment:
The requirement is that a stored AnonCreds W3C VC format credential must be accessible in the same way as a legacy AnonCreds VC format -- searchable in the same way.  In storing credentials, encrypted tags are generated to enable querying for a credential (using [WQL]) to find ones in a holder's storage that satisfy a presentation request. I'm not sure of the implication of that. Does an AnonCreds VC in W3C VC Format need to be converted to legacy format before storage?

I'll see what I can find out about that.

Reply:
I'm not sure about changes that Askar might have made, but I'm pretty sure that in the original wallet impls, the cred itself was just an opaque blob, so it wouldn't need a format conversion before storing. What we would need, however, is a comparable extraction of metadata (cred def, schema, issuer, date, etc).

Reply:
Agree — I’ll see what I can find out about where/how that happens.